server/rest/src/main/java/org/infinispan/rest/resources/CounterResource.java

```diff
@@ -30,84 +50,299 @@
+            .invocation().methods(GET).path("/v2/counters/{counterName}").withAction("reset").handleWith(this::resetCounter)
+
+            // Weak counter ops
+            .invocation().methods(GET).path("/v2/counters/{counterName}").withAction("increment").handleWith(this::incrementWeak)
+            .invocation().methods(GET).path("/v2/counters/{counterName}").withAction("decrement").handleWith(this::decrementWeak)
+            .invocation().methods(GET).path("/v2/counters/{counterName}").withAction("add").handleWith(this::addValueWeak)
+
+            // Strong counter ops
+            .invocation().methods(GET).path("/v2/counters/{counterName}").withAction("incrementAndGet").handleWith(this::incrementStrong)
+            .invocation().methods(GET).path("/v2/counters/{counterName}").withAction("decrementAndGet").handleWith(this::decrementStrong)
+            .invocation().methods(GET).path("/v2/counters/{counterName}").withAction("addAndGet").handleWith(this::addValueStrong)
+            .invocation().methods(GET).path("/v2/counters/{counterName}").withAction("compareAndSet").handleWith(this::compareSet)
+            .invocation().methods(GET).path("/v2/counters/{counterName}").withAction("compareAndSwap").handleWith(this::compareSwap)
             .create();
    }
 
+   private CompletionStage<RestResponse> createCounter(RestRequest request) throws RestResponseException {
+      NettyRestResponse.Builder responseBuilder = new NettyRestResponse.Builder();
+      String counterName = request.variables().get("counterName");
+
+      String contents = request.contents().asString();
+
+      if (contents == null || contents.length() == 0) {
+         responseBuilder.status(HttpResponseStatus.BAD_REQUEST);
+         responseBuilder.entity("Configuration not provided");
+         return completedFuture(responseBuilder.build());
+      }
+      try {
+         CounterConfiguration configuration = createCounterConfiguration(contents);
+         if (configuration == null) {
+            responseBuilder.status(HttpResponseStatus.BAD_REQUEST);
+            responseBuilder.entity("Invalid configuration");
+            return completedFuture(responseBuilder.build());
+         }
+         return counterManager.defineCounterAsync(counterName, configuration).thenApply(r -> new NettyRestResponse.Builder().build());
+      } catch (IOException e) {
+         throw new RestResponseException(e.getCause());
+      }
+   }
+
+   private CounterConfiguration createCounterConfiguration(String json) throws IOException {
+      JsonNode jsonNode = invocationHelper.getMapper().readValue(json, JsonNode.class);
+      boolean strongCounter = jsonNode.has("strong-counter");
+      boolean weakCounter = jsonNode.has("weak-counter");
+
+      GlobalConfiguration cacheManagerConfiguration = invocationHelper.getRestCacheManager().getInstance().getCacheManagerConfiguration();
+      GlobalConfigurationBuilder builder = new GlobalConfigurationBuilder();
```

Comment:
you can use `CounterConfiguration.builder(<type>)` to build the counter config without going through the `CounterManagerConfiguration`

Reply:
I need an instance of either ```StrongCounterConfigurationBuilder``` or ```WeakCounterConfigurationBuilder``` to read a JSON, since those are the classes where the Attributes are defined

Reply:
Maybe there's an easier way to do it?

Reply:
well, if you want to use the json-reader, you can use `StrongCounterConfigurationBuilder(null)`. 

the `CounterManagerConfigurationBuilder` is only used to validate but the `CounterManager` does the validation for you.

Reply:
ok, that makes sense, let me try it